lib/ansible/module_utils/basic.py

```diff
@@ -524,6 +524,35 @@ def is_executable(path):
             or stat.S_IXGRP & os.stat(path)[stat.ST_MODE]
             or stat.S_IXOTH & os.stat(path)[stat.ST_MODE])
 
+def get_remote_temp():
+
+    ''' Return the remote temp path where modules are stored '''
+
+    # If using test-module and explode, the remote temp will resemble ...
+    #   /tmp/test_module_scratch/debug_dir/ansible/module_utils/basic.py
+    # If using ansible or ansible-playbook with a remote system ...
+    #   /tmp/ansible_vmweLQ/ansible_modlib.zip/ansible/module_utils/basic.py
+
+    # the path to basic.py
+    thisfile = __file__
+
+    remote_temp = None
+
+    # list.index('ansible') in reverse
+    directories = thisfile.split(os.sep)
+    directories = [x for x in reversed(directories)]
+    ansible_index = None
+    for idx,x in enumerate(directories):
+        if x == 'ansible':
+            ansible_index = idx
+            break
+    if ansible_index:
+        # Keep everything up to the index
+        directories = directories[(ansible_index + 1):]        
+        directories = [x for x in reversed(directories)]
+        remote_temp = os.sep.join(directories)
+    return remote_temp
+ 
```

Comment:
I think this may need to be more paranoid.  I think as it is it will break if controller code runs from a checkout and makes use of this remote_temp() method. Maybe we should just check for ```.endswith("debug_dir/ansible/module_utils/basic.py") or .endswith("ansible_modlib.zip/ansible/module_utils/basic.py")```?